src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -95,7 +97,7 @@ class MemAllocator::Allocation::PreserveObj: StackObj {
   oop* const _obj_ptr;
 
 public:
-  PreserveObj(JavaThread* thread, oop* obj_ptr)
+  PreserveObj(Thread* thread, oop* obj_ptr)
```

Comment:
Why did you change this? Following through the usage call chain it is only used by the `_thread` from `Allocation` which has to be a `JavaThread`.

Reply:
I wanted to minimize the use of `JavaThread`. Most of the MemAllocator APIs take `Thread`, so any occurrence of `JavaThread` looks suspicious and may cause people to unwittingly cast from `Thread` to `JavaThread` in the future.

Reply:
Sometimes MemAllocator is used by non-Java threads (e.g., GC threads uses it to create filler objects).

Reply:
I reverted the change in `PreserveObj`, which is a private inner class of `MemAllocator::Allocation`, so the use of `JavaThread*` doesn't leak to the outside world.